index.bs

```diff
@@ -1130,6 +1166,314 @@ The following extended attributes are applicable to interfaces:
+that declares state and behavior that can be [=included=] by one or more [=interfaces=],
+and that are exposed by objects that implement an [=interface=] that [=includes=] the [=mixin=].
+
+<pre class="webidl" class="syntax">
+    interface mixin identifier {
+      /* mixin_members... */
+    };
+</pre>
+
+Note: [=Mixins=], much like [=partial interfaces=],
+are intended for use as a specification editorial aide,
+allowing a coherent set of functionalities to be grouped together,
+and included in multiple interfaces, possibly across documents.
+They are not meant to be exposed through language bindings.
+Guidance on when to choose [=partial interface|partials=], [=mixins=], or [=partial mixins=]
+can be found in the <a href="#when-to-use-partials-vs-mixins">following note</a>.
+
+A [=mixin=] is a specification of a set of <dfn export lt="mixin member">mixin members</dfn>
+(matching <emu-nt><a href="#prod-MixinMembers">MixinMembers</a></emu-nt>),
+which are the [=constants=], [=regular operations=], [=regular attributes=], and [=stringifiers=]
+that appear between the braces in the [=mixin=] [=declaration=].
+
+These [=constants=], [=regular operations=], [=regular attributes=], and [=stringifiers=]
+describe the behaviors that can be implemented by an object,
+as if they were specified on the [=interface=] that [=includes=] them.
+
+[=Static attributes=], [=static operations=], [=special operations=], except for [=stringifiers=], and
+[=iterable declaration|iterable=], [=maplike declaration|maplike=], and [=setlike declarations=]
+cannot appear in [=mixin=] [=declarations=].
+
+As with interfaces, the IDL for [=mixins=] can be split into multiple parts by using
+<dfn export>partial mixin</dfn> definitions
+(matching <emu-t>partial</emu-t> <emu-t>interface</emu-t> <emu-nt><a href="#prod-MixinRest">MixinRest</a></emu-nt>).
+The [=identifier=] of a [=partial mixin=] [=definition=] must
+be the same as the [=identifier=] of a [=mixin=] [=definition=].
+All of the [=mixin members|members=] that appear on each of the [=partial mixin=] [=definitions=]
+are considered to be [=members=] of the [=mixin=] itself,
+and—by extension—of the [=interfaces=] that [=include=] the [=mixin=].
+
+<pre class="webidl" class="syntax">
+    interface mixin <mark>SomeMixin</mark> {
+      /* mixin_members... */
+    };
+
+    partial interface mixin <mark>SomeMixin</mark> {
+      /* mixin_members... */
+    };
+</pre>
+
+The order that members appear in has significance for property enumeration
+in the <a href="#es-namespaces">ECMAScript binding</a>.
+
+Note that unlike [=interfaces=] or [=dictionaries=], [=mixins=] do not create types.
+
+Of the extended attributes defined in this specification,
+only the [{{Exposed}}] and [{{SecureContext}}] extended attributes are applicable to [=mixins=].
+
+An <dfn>includes statement</dfn> is a definition
```

Comment:
Maybe we should have two subsections, "Includes statements" and "Using mixins and partials"? (For the latter, it would become a non-normative section, instead of a green-background note.)

Reply:
Turned the second note into an informative section. Had to move it to the end of the section however, if not the grammar that's tagged on the end feels out of place. Will be tackling this further in #450.